pushbullet

```diff
@@ -106,28 +119,57 @@ function getactivepushes () {
 	done
 }
 
+checkCurlReturnCode() {
+	errcode=${1:-$ERR_UNK}
+	if [ "${errcode}" = "${ERR_UNK}" ]; then
+		err "Unknown error (${errcode})"
+		exit ${errcode}
+	elif [ "${1}" != "0" ]; then
+		err "Unknown error from curl (${1})"
+		exit ${1}
+	fi
+}
+
 checkCurlOutput() {
 	res=$(echo "$1" | grep -o "created" | tail -n1)
-	if [[ "$1" == *"The param 'channel_tag' has an invalid value."* ]] && [[ "$1" == *"The param 'device_iden' has an invalid value."* ]]; then
-		err "Error: You specified an unknown device or channel."
-		exit 1
-	elif [[ "$1" == *"invalid_access_token"* ]]; then
-		err "Access token is missing or invalid."
-		exit 1
-	elif [[ "$1" == *"Object not found"* ]]; then
-		err "Object not found"
-		exit 1
-	elif [[ "$1" == '{"accounts":[],"blocks":[],"channels":[],"chats":[],"clients":[],"contacts":[],"devices":[],"grants":[],"pushes":[],"profiles":[],"subscriptions":[],"texts":[]}' ]]; then
-		# "empty" response
-		return 0
-	elif [[ "$1" == '{"accounts":[],"blocks":[],"channels":[],"chats":[],"clients":[],"contacts":[],"devices":[],"grants":[],"pushes":[],"profiles":[],"subscriptions":[],"texts":[],"cursor":'* ]]; then
-		# only next cursor, otherwise empty response
-		# suppress error when calling break outside of loop
-		break 2&> /dev/null
-	elif [[ "$res" != "created" ]] && [[ ! "$1" == "{}" ]]; then
-		err "Error submitting the request. The error message was:" "$1"
-		exit 1
-	fi
+	curlretcode=${2:-$ERR_UNK}
+	checkCurlReturnCode "$curlretcode"
+	case "$1" in
+		*"The param 'channel_tag' has an invalid value."*)
```

Comment:
while this is a correct translation of the code, I think this is not that the original intention was. I think instead this should be
```
case ...
*"The param 'channel_tag' has an invalid value."* | *"The param 'device_iden' has an invalid value."*)
...